### PR TITLE
[RISC-V] Smarter update kernel part of user top PD

### PR DESCRIPTION
**Pull request by cahirwpz**
This is an issue on RISC-V where we must update upper part of each user pmap if top page directory of kernel pmap was modified.

### CODE REVIEW

File: sys/riscv/pmap.c [hunk 1]
```diff
@@ -84,27 +84,71 @@ inline pte_t pte_protect(pte_t pte, vm_prot_t prot) {
...
   __set_satp(umap->md.satp);
   __sfence_vma();
 }
 
+void pmap_md_update(pmap_t *umap) {
+  if (!umap)
+    return;
+
+  update_kernel_pd(umap);
+  __sfence_vma();
```
Comment (MichalBlk) [inline sys/riscv/pmap.c:149]: If the generation is up to date, then the `__sfence_vma` is needless. You could move this statement after the `memcpy` invocation.

File: sys/riscv/pmap.c [hunk 2]
```diff
@@ -84,27 +84,71 @@ inline pte_t pte_protect(pte_t pte, vm_prot_t prot) {
...
+      *pdep = pde_make(0, kmap_l0_pages[L0_INDEX(va)]);
+    }
+
+    kmap->md.generation++;
+    assert(kmap->md.generation);
+  }
+}
+
+void pmap_md_activate(pmap_t *umap) {
+  update_kernel_pd(umap);
```
Comment (MichalBlk) [inline sys/riscv/pmap.c:138]: We should consider the case when `umap` is `NULL` (i.e. switching to a kernel thread). Currently, this isn't resulting in a kernel oops, as we don't update `curpmap` correctly (see #1353). You can keep it as it is, and I'll update #1353 after this PR is merged. Or merge that PR first and update this procedure.

**Review by MichalBlk — APPROVED** [review]
LGTM. Please add to the description an explanation regarding the lock cycle (similar to description in #1336).

File: sys/riscv/pmap.c (hunk 1 above)
Comment (cahirwpz) [reply sys/riscv/pmap.c:149]: Done!

File: sys/riscv/pmap.c (hunk 2 above)
Comment (cahirwpz) [reply sys/riscv/pmap.c]: Done!

**Comment by cahirwpz** [comment]
@MichalBlk I added a comment on why `kmap_l0_lock` is needed.

**Pull request merged**